File: akka-javasdk/src/main/java/akka/javasdk/agent/autonomous/AgentCapability.java
/*
 * Copyright (C) 2021-2026 Lightbend Inc. <https://www.lightbend.com>
 */

package akka.javasdk.agent.autonomous;

import java.util.List;

/** Internal representation of agent capabilities. */
sealed interface AgentCapability permits AgentCapability.Delegation, AgentCapability.Handoff {

  record Delegation(List<Class<? extends AutonomousAgent>> agents) implements AgentCapability {}

  record Handoff(List<Class<? extends AutonomousAgent>> agents) implements AgentCapability {}
}

File: akka-javasdk/src/main/scala/akka/javasdk/impl/agent/autonomous/capability/AgentCapability.scala
/*
 * Copyright (C) 2021-2026 Lightbend Inc. <https://www.lightbend.com>
 */

package akka.javasdk.impl.agent.autonomous.capability

import akka.annotation.InternalApi

/**
 * INTERNAL API
 *
 * Common trait for all agent capability types. Capabilities are stored in the agent definition or setup and converted
 * to SPI types by [[akka.javasdk.impl.agent.autonomous.CapabilityConverter]].
 */
@InternalApi
private[javasdk] trait AgentCapability
